### DISCUSSION TITLE
Need for benchmarks to deliver performance optimizations

### DISCUSSION THREAD

**Discussion by k06a** [paradigmxyz/reth#1066] [Technical discussions]
### Describe the feature

I see `README.md` states:

2. Performance: Reth aims to be fast, so we used Rust and the [Erigon staged-sync](https://erigon.substack.com/p/erigon-stage-sync-and-control-flows) node architecture. We also use our Ethereum libraries (including [ethers-rs](https://github.com/gakonst/ethers-rs/) and [revm](https://github.com/bluealloy/revm/)) which we’ve battle-tested and optimized via [Foundry](https://github.com/paradigmxyz/foundry/).

I assume this means project need to have exhaustive benchmarking of different stages of sync/run and benchmarking documentation, this would allow anyone to contribute performance optimisations.

**Comment by onbjerg** [comment]
I think this is a bit too broad of an issue, so I would like this to be converted into a tracking issue with actionable tasks. For that reason I will convert this to a discussion first, with the reason being that the issue tracker should be as actionable as possible.

We do have plans to add benchmarks for the individual stages using database snapshots, as well as benchmarks for running the node fully to sync over specific block ranges
